traveler-ios-ui/TravelerKitUI/ViewControllers/WishlistResultViewController.swift

```diff
@@ -0,0 +1,241 @@
+//
+//  WishlistResultViewController.swift
+//  TravelerKitUI
+//
+//  Created by Ben Ruan on 2019-09-04.
+//  Copyright © 2019 GuestLogix. All rights reserved.
+//
+
+import UIKit
+import TravelerKit
+
+let wishlistCellIdentifier = "wishlistCellIdentifier"
+let loadingCollectionCellIdentifier = "loadingCellIdentifier"
+
+public protocol WishlistResultViewControllerDelegate: class {
+    func wishlistResultViewControllerDidRefresh(_ controller: WishlistResultViewController)
+}
+
+open class WishlistResultViewController: UICollectionViewController {
+    private var _volatileResult: WishlistResult?
+    private var selectedCatalogItem: CatalogItem?
+    private var pagesLoading = Set<Int>()
+
+    public weak var delegate: WishlistResultViewControllerDelegate?
+
+    public var wishlistResult: WishlistResult?
+
+    override open func viewDidLoad() {
+        super.viewDidLoad()
+
+        _volatileResult = wishlistResult
+    }
+
+    override open func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        switch (segue.identifier, segue.destination) {
+        case (_, let navVC as UINavigationController):
+            let vc = navVC.topViewController as? CatalogItemViewController
+            vc?.catalogItem = selectedCatalogItem
+        default:
+            Log("Unknown segue", data: segue, level: .warning)
+            break
+        }
+    }
+
+    // MARK: UICollectionViewDataSource
+
+    override open func numberOfSections(in collectionView: UICollectionView) -> Int {
+        return 1
+    }
+
+    override open func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return wishlistResult?.total ?? 0
+    }
+
+    override open func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let catalogItem = wishlistResult!.items[indexPath.row]
+
+        switch catalogItem {
+        case .none:
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: loadingCollectionCellIdentifier, for: indexPath)
+            cell.contentView.subviews.forEach({ $0.startShimmering() })
+            return cell
+        case .some(let item):
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: wishlistCellIdentifier, for: indexPath) as! WishlistCell
+
+            cell.titleLabel.text = item.title
+            cell.descriptionLabel.text = item.subTitle
+            cell.priceLabel.text = "Price starting at: " + (item.price.localizedDescriptionInBaseCurrency ?? "")
+            cell.isAvailable = item.isAvailable
+
```

Comment:
You should set the imageviews image to nil immediately and then do AssetManager call

This will prevent stale images from being shown